libosrmc/osrmc.cc

```diff
@@ -188,15 +211,34 @@ float osrmc_route_response_duration(osrmc_route_response_t response, osrmc_error
   const auto duration = route.values["duration"].get<osrm::json::Number>().value;
   return duration;
 } catch (const std::exception& e) {
-  *error = new osrmc_error{e.what()};
+  osrmc_error_from_exception(e, error);
   return INFINITY;
 }
 
+osrmc_table_annotations_t osrmc_table_annotations_construct(osrmc_error_t* error) try {
+  auto* out = new osrm::TableParameters::AnnotationsType{osrm::TableParameters::AnnotationsType::Duration};
+  return reinterpret_cast<osrmc_table_annotations_t>(out);
+} catch (const std::exception& e) {
+  osrmc_error_from_exception(e, error);
+  return nullptr;
+}
+
+void osrmc_table_annotations_destruct(osrmc_table_annotations_t annotations) {
+  delete reinterpret_cast<osrm::TableParameters::AnnotationsType*>(annotations);
+}
+
+void osrmc_table_annotations_set_distance(osrmc_table_annotations_t annotations, osrmc_error_t* error) try {
```

Comment:
Do you think we should provide a way to set and unset a specific annotation? As in an API like `enable_distance(true)` or `enable_distance(false)` for setting and unsetting the distance annotation.